include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -705,7 +705,7 @@ struct ignore_copyable
     }
 
     bool
-    operator==(const ignore_copyable& other) const
+    operator==(const ignore_copyable&) const
```

Comment:
may be add == delete ? Or simple remove this operator ?

Reply:
In this case we must delete `operator!=` too and then modify `iterators.pass.cpp` not to run `*it == *it` scenario for `discard_iterator`. 
I am not sure if `*it == *it` or `*it != *it` have any sense for `discard_iterator`, but giving up their support will break backward compatibility.

@MikeDvorskiy could you comment regarding keeping this operations and usage scenarios of `*it == *it` for discard iterator?